examples/effects-and-handlers/Black-Jack.flix

```diff
@@ -0,0 +1,178 @@
+///
+/// Player Actions.
+///
+enum Action with ToString {
+    case Hit,
+    case Stand
+}
+
+///
+/// Player Action.
+///
+eff Play {
+    def hitOrStand(): Result[String,Action]
+}
+
+///
+/// Card Ranks.
+///
+enum Card with ToString, Eq {
+    case Two,
+    case Three,
+    case Four,
+    case Five,
+    case Six,
+    case Seven,
+    case Eight,
+    case Nine,
+    case Ten,
+    case Jack,
+    case Queen,
+    case King,
+    case Ace
+}
+
+///
+/// Return the numeric value of a Card.
+///
+def cardValue(card: Card): Int32 = match card {
+   case Card.Two => 2
+   case Card.Three => 3
+   case Card.Four => 4
+   case Card.Five => 5
+   case Card.Six => 6
+   case Card.Seven => 7
+   case Card.Eight => 8
+   case Card.Nine => 9
+   case Card.Ace => 11
+   case _ => 10 // 10 and face cards
+}
+
+///
+/// Pop a card from the top of the Deck.
+///
+def pullCard(deck: List[Card]): Option[(Card, List[Card])] \ IO = match deck {
```

Comment:
In an effect-oriented program, I would expect IO, NonDet, etc. only to show up in the handlers.

IO would be delegated to some user interface effect, which could be implemented either with the console or graphical UI.